core/src/main/java/org/apache/calcite/sql/fun/SqlAnyValueAggFunction.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql.fun;
+
+import org.apache.calcite.sql.SqlAggFunction;
+import org.apache.calcite.sql.SqlFunctionCategory;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.type.OperandTypes;
+import org.apache.calcite.sql.type.ReturnTypes;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * Definition of the <code>ANY_VALUE</code> aggregate functions,
+ * returning any one of the values which go into it.
+ *
+ * <p>There are 3 forms:
+ *
+ * <dl>
+ * <dt>sum(<em>primitive type</em>)
+ * <dd>values are compared using '&lt;'
+ *
+ * <dt>sum({@link java.lang.Comparable})
+ * <dd>values are compared using {@link java.lang.Comparable#compareTo}
+ *
+ * <dt>sum({@link java.util.Comparator}, {@link java.lang.Object})
+ * <dd>the {@link java.util.Comparator#compare} method of the comparator is used
+ * to compare pairs of objects. The comparator is a startup argument, and must
+ * therefore be constant for the duration of the aggregation.
+ * </dl>
+ */
+public class SqlAnyValueAggFunction extends SqlAggFunction {
+
+  //~ Instance fields --------------------------------------------------------
+
+  //~ Constructors -----------------------------------------------------------
+
+  /** Creates a SqlAnyValueAggFunction. */
+  public SqlAnyValueAggFunction(SqlKind kind) {
+    super(kind.name(),
+        null,
+        kind,
+        ReturnTypes.ARG0_NULLABLE_IF_EMPTY,
+        null,
+        OperandTypes.COMPARABLE_ORDERED,
```

Comment:
Is there a reason for making this comparable_ordered ?

Reply:
No, changed it to ANY.